Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/sheets/person/ContactDiaryPersonBottomSheetDialogViewModel.kt

```diff
@@ -0,0 +1,33 @@
+package de.rki.coronawarnapp.contactdiary.ui.day.sheets.person
+
+import com.squareup.inject.assisted.AssistedInject
+import de.rki.coronawarnapp.contactdiary.storage.entity.ContactDiaryPersonEntity
+import de.rki.coronawarnapp.contactdiary.storage.repo.ContactDiaryRepository
+import de.rki.coronawarnapp.ui.SingleLiveEvent
+import de.rki.coronawarnapp.util.coroutine.DispatcherProvider
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModel
+import de.rki.coronawarnapp.util.viewmodel.SimpleCWAViewModelFactory
+
+class ContactDiaryPersonBottomSheetDialogViewModel @AssistedInject constructor(
+    dispatcherProvider: DispatcherProvider,
+    private val contactDiaryRepository: ContactDiaryRepository
+) : CWAViewModel(dispatcherProvider = dispatcherProvider) {
+
+    val shouldClose = SingleLiveEvent<Unit>()
+
+    fun savePerson(fullName: String) = launch {
```

Comment:
personNames are only allowed with max(250chars)
could be handled in different subtask, but maybe here :-)

Reply:
added validation and ui hints for field constraints